internal/v4/relations.go

```diff
@@ -222,8 +227,12 @@ func (h *Handler) metaBundlesContaining(entity *mongodoc.Entity, id *charm.Refer
 		if err != nil {
 			return nil, errgo.Notef(err, "cannot retrieve bundle metadata")
 		}
+		url := e.URL
+		if e.PromulgatedURL != nil {
```

Comment:
I had one of those and took it out. often the URL that is chosen is based on the request that is made rather than what is available.